### PR TITLE
chore(eslint): update eslint to latest config

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

* Updated eslint, eslint-config-algolia and babel-eslint to the newest version
* add babel to the browserify build step

**Result**

lots of changes, some ignored rules and a test suite that maybe does not pass anymore

### CODE REVIEW

**Comment by vvo** [comment]
Did you check that IE8 is able to run the examples?

File: examples/parse.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-var algoliasearch = require('cloud/algoliasearch.parse.js');
-var client = algoliasearch('latency', '6be0576ff61c053d5f9a3225e2a90f76');
-var index = client.initIndex('contacts');
+const algoliasearch = require('cloud/algoliasearch.parse.js');
+const client = algoliasearch('latency', '6be0576ff61c053d5f9a3225e2a90f76');
+const index = client.initIndex('contacts');
 
-Parse.Cloud.define('hello', function(request, response) {
-  index.search('Atlenta', function(err, results) {
+Parse.Cloud.define('hello', (request, response) => {
```
Comment (vvo) [inline examples/parse.js]: Unsure arrow methods works on Parse

File: scripts/build [hunk 2]
```diff
@@ -4,7 +4,7 @@ set -e # exit when error
 # Use this with `npm run build` so that `./node_modules/.bin` are injected into
 # the PATH
 
-license="/*! algoliasearch ${VERSION:-UNRELEASED} | © 2014, 2015 Algolia SAS | github.com/algolia/algoliasearch-client-js */"
+license="/*! algoliasearch ${VERSION:-UNRELEASED} | © 2014 - present Algolia Inc. | github.com/algolia/algoliasearch-client-js */"
```
Comment (vvo) [inline scripts/build]: It's Algolia, Inc. I think

**Comment by vvo** [comment]
Did you check that it's also working on the currently supported Node.js engines? I think we do support Node >= 0.10

**Comment by Haroenv** [comment]
Works in IE8

**Comment by vvo** [comment]
For Node.js you might have to compile too

**Comment by Haroenv** [comment]
Should I run the test or the examples in node?

**Comment by vvo** [comment]
@Haroenv the tests should pass on Node 0.10 but you can also just do node examples/node

**Comment by vvo** [comment]
Also overall if you are starting fresh, this might have been a good experiment but not something you absolutely want to get merged. Your call

**Comment by Haroenv** [comment]
Good call, I'll close this for now, and work on the fresh things.

**Pull request closed without merging**